launcher/src/main/java/com/threerings/getdown/launcher/Getdown.java

```diff
@@ -507,13 +479,13 @@ protected void updateJava ()
         vmjar.install(true);
 
         // these only run on non-Windows platforms, so we use Unix file separators
-        String localJavaDir = LaunchUtil.LOCAL_JAVA_DIR + "/";
-        FileUtil.makeExecutable(_app.getLocalPath(localJavaDir + "bin/java"));
-        FileUtil.makeExecutable(_app.getLocalPath(localJavaDir + "lib/jspawnhelper"));
-        FileUtil.makeExecutable(_app.getLocalPath(localJavaDir + "lib/amd64/jspawnhelper"));
+        File javaLocalDir = _app.getJavaLocalDir();
+        FileUtil.makeExecutable(new File(javaLocalDir, "bin/java"));
+        FileUtil.makeExecutable(new File(javaLocalDir, "lib/jspawnhelper"));
+        FileUtil.makeExecutable(new File(javaLocalDir, "lib/amd64/jspawnhelper"));
```

Comment:
i wonder. getdown works intensively with file pathds and currently requires java 1.7 so why not use Paths API?

Reply:
Getdown's code was written when Java 1.5 was state of the art. It only recently switched to requiring Java 1.7 as a minimum version. At that time I didn't have the time nor inclination to go rewrite a bunch of code that had been working fine for 15 years just because we could now use a new API for dealing with file paths. I like the `nio` APIs, but I don't like making major changes to the code base just for fun.